WireAPI/Sources/WireAPI/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,156 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireFoundation
+
+// sourcery: AutoMockable
+protocol AuthenticationManagerProtocol {
+
+    func getValidAccessToken() async throws -> AccessToken
+    func refreshAccessToken() async throws -> AccessToken
+
+}
+
+actor AuthenticationManager: AuthenticationManagerProtocol {
+
+    private var currentToken: CurrentToken?
+    private let clientID: String
+    private let cookieStorage: CookieStorage
+    private let networkService: NetworkService
+
+    init(
+        clientID: String,
+        cookieStorage: CookieStorage,
+        networkService: NetworkService
+    ) {
+        self.clientID = clientID
+        self.cookieStorage = cookieStorage
+        self.networkService = networkService
+    }
+
+    func getValidAccessToken() async throws -> AccessToken {
+        switch currentToken {
+        case .renewing(let task):
+            return try await task.value
+
+        case .cached(let accessToken) where !accessToken.isExpiring:
```

Comment:
OK so the access token is only kept in memory? Is that correct?

Reply:
Correct